### ISSUE TITLE
Using #[inline] on function prototype should emit a warning

### ISSUE THREAD

**Issue by Havvy** [rust-lang/rust#51280]
The following trait does not emit any warnings, although it should warn that the `#[inline]` hint is ignored on function prototypes.

```rust
trait Trait {
    #[inline]
    fn function();
}
```

Likewise for extern functions. E.g. should warn on

```rust
extern {
  #[inline] fn function();
}
````

**Comment by jonas-schievink** [comment]
Your trait method is a default method - are inline hints actually ignored there, too (that sounds like a bug)? Perhaps you meant to declare it without a body.

**Comment by Havvy** [comment]
Yes, meant to declare without a body. Example fixed.

**Comment by keringar** [comment]
If anyone's interested in pushing for this by default, it can already be found in clippy. https://rust-lang-nursery.github.io/rust-clippy/master/index.html#inline_fn_without_body
